### PR TITLE
Use new types in solo 2

### PR DESCRIPTION
**Pull request by vitaliy-io**

### CODE REVIEW

File: packages/solo/ledgerl1l2.go [hunk 1]
```diff
@@ -71,80 +72,91 @@ func (ch *Chain) L2Assets(agentID isc.AgentID) *isc.Assets {
 func (ch *Chain) L2AssetsAtStateIndex(agentID isc.AgentID, stateIndex uint32) *isc.Assets {
```
Comment (dessaya) [inline packages/solo/ledgerl1l2.go]: This should return `CoinBalances` instead of `Assets` IMO. As a general rule, if a function only deals with coins and not NFTs/objects, it should deal with `CoinBalances`.

File: packages/solo/ledgerl1l2.go [hunk 2]
```diff
@@ -29,15 +30,15 @@ func (ch *Chain) L2Accounts() []isc.AgentID {
...
-}
+// func (ch *Chain) parseAccountBalance(d dict.Dict, err error) *isc.Assets {
+// 	require.NoError(ch.Env.T, err)
+// 	if d.IsEmpty() {
+// 		return isc.NewEmptyAssets()
+// 	}
+// 	ret, err := isc.AssetsFromDict(d)
+// 	require.NoError(ch.Env.T, err)
+// 	return ret
+// }
```
Comment (dessaya) [inline packages/solo/ledgerl1l2.go]: It's ok to delete this

File: packages/solo/ledgerl1l2.go [hunk 3]
```diff
@@ -71,80 +72,91 @@ func (ch *Chain) L2Assets(agentID isc.AgentID) *isc.Assets {
...
+// 	return out.(*iotago.FoundryOutput), nil
+// }
+
+// func (ch *Chain) GetNativeTokenIDByFoundrySN(sn uint32) (iotago.NativeTokenID, error) {
+// 	o, err := ch.GetFoundryOutput(sn)
+// 	if err != nil {
+// 		return iotago.NativeTokenID{}, err
+// 	}
+// 	return o.MustNativeTokenID(), nil
+// }
```
Comment (dessaya) [inline packages/solo/ledgerl1l2.go]: ok to delete

File: packages/vm/core/accounts/state.go [hunk 4]
```diff
@@ -48,7 +48,7 @@ func NewStateWriterFromSandbox(ctx isc.Sandbox) *StateWriter {
 }
 
 // converts an account key from the accounts contract (shortform without chainID) to an AgentID
-func agentIDFromKey(key kv.Key, chainID isc.ChainID) (isc.AgentID, error) {
+func AgentIDFromKey(key kv.Key, chainID isc.ChainID) (isc.AgentID, error) {
```
Comment (dessaya) [inline packages/vm/core/accounts/state.go:51]: Why this change?

**Review by dessaya — APPROVED** [review]

File: packages/solo/ledgerl1l2.go (hunk 1 above)
Comment (vitaliy-io) [reply packages/solo/ledgerl1l2.go]: done

File: packages/solo/ledgerl1l2.go (hunk 2 above)
Comment (vitaliy-io) [reply packages/solo/ledgerl1l2.go]: done

File: packages/vm/core/accounts/state.go (hunk 4 above)
Comment (vitaliy-io) [reply packages/vm/core/accounts/state.go:51]: I needed to create AgentID in L2Accounts() func in ledgerl1l2.go

File: packages/solo/ledgerl1l2.go (hunk 3 above)
Comment (vitaliy-io) [reply packages/solo/ledgerl1l2.go]: done
